### ISSUE TITLE
Server rejectes as Unsupported media type because Content-Type is missing for POST

### ISSUE THREAD

**Issue by konczak** [Orange-OpenSource/angular-swagger-ui#10]
Hi,
I've checked version 0.1.3 and 0.1.6 and in both have the same problem.
When using try-it-out button for POST or PUT request is send but Content-Type and body are missing (checked on both client and server).
In result server rejects request with status HTTP 415 Unsupported media type.

I tried to force Content-Type to be application/json using custom transform like:

``` javascript
if (request.method === 'post'
        || request.method === 'put') {
   request.headers['Content-Type'] = 'application/json';
}
```

But it didn't help at all. While actually overriding Authorization works as expected.
Code compliant with version 0.1.6 looks like that:

``` html
<div swagger-ui=""
     url="swaggerUrl"
     api-explorer="true"
     error-handler="myErrorHandler"
     transform-request="myTransform"
     try-it="true">
</div>
```

``` javascript
(function () {
   'use strict';

   angular
           .module('appraisalOnlineApp')
           .controller('DocumentationApiCtrl', DocumentationApiCtrl);

   DocumentationApiCtrl.$inject = ['$scope', 'toaster', 'logToServerUtil'];

   function DocumentationApiCtrl($scope, toaster, logToServerUtil) {
      $scope.businessApi = 'v2/api-docs?group=Business-API';
      $scope.adminApi = 'v2/api-docs?group=Admin-API';
      $scope.authorization = 'blabla';
      $scope.myErrorHandler = myErrorHandler;
      $scope.myTransform = myTransform;
      $scope.switchApi = switchApi;

      // init form
      switchApi($scope.businessApi);

      //////////

      function switchApi(api) {
         $scope.url = api;
         $scope.swaggerUrl = api;
      }
      // error management
      function myErrorHandler(data, status) {
         logToServerUtil.error('Load Swagger documentation failure', status);
         toaster.pop('error', null, 'documentation/api/load-failure', null, 'template');
      }
      // transform API explorer requests
      function myTransform(request) {
         console.log('request', request);
         if (request.method === 'post'
                 || request.method === 'put') {
            console.log('yes sir');
            request.headers['Content-Type'] = 'application/json';
         }
         request.headers['Authorization'] = $scope.authorization;
      }
   }
})();

```

**Comment by mathieuales** [comment]
Hi !

Thanks for using angular-swagger-ui.
It's a documentation issue :-( when using 0.1.6 you should use directive as described below to enable request transformation:

```
<div swagger-ui
     url="swaggerUrl"
     api-explorer="true"
     error-handler="myErrorHandler"
     api-explorer-transform="myTransform">
</div>
```

Then make sure your swagger API descriptor defines the Content-Type of your operation in its definition [see consumes parameter](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#operation-object)

You can also take a look at the example in dist/index.html.

Hope this helps !

**Comment by konczak** [comment]
I think that integration of Angular-Swagger-Bootstrap is awesome idea - thank you for your work!

You are right that I should use api-explorer-transform in 0.1.6 so I change it but issue still remain the same. 

I know that Content-Type served by server has to be in place and it is as you can see:

``` json
 "post":{
    "tags":[
       "place-api"
    ],
    "summary":"Add new Place [HR]",
    "description":"Returns created Place",
    "operationId":"addUsingPOST",
    "consumes":[
       "application/json"
    ],
    "produces":[
       "*/*"
    ],
    "parameters":[
       {
          "in":"body",
          "name":"placeNew",
          "description":"placeNew",
          "required":true,
          "schema":{
             "$ref":"#/definitions/PlaceNew"
          }
       }
    ],
    "responses":{
       "200":{
          "description":"Create new Place",
          "schema":{
             "$ref":"#/definitions/Place"
          }
       },
       "201":{
          "description":"Created"
       },
       "400":{
          "description":"Invalid input"
       },
       "401":{
          "description":"Unauthorized"
       },
       "403":{
          "description":"Forbidden"
       },
       "404":{
          "description":"Not Found"
       }
    }
 }
```

And you can see application/json content in the UI to choose and that body is specified as well:
![zrzut ekranu 2015-07-14 12 37 27](https://cloud.githubusercontent.com/assets/1986726/8671408/20843284-2a25-11e5-9969-7f186f5d68c2.png)

**Comment by mathieuales** [comment]
I found the issue, it ill be fixed in next release ASAP.
Workaround: all parameters in body must be named "body" in your Swagger API descriptor.

**Comment by mathieuales** [comment]
should be fixed in 0.2
can you confirm ?
